src/instruction/math/and.rs

```diff
@@ -0,0 +1,100 @@
+use instruction::instruction::ExecuteResult;
+use rtda::frame::Frame;
+use util::code_reader::CodeReader;
+
+#[allow(non_snake_case)]
+pub fn IAND(code_reader: CodeReader, frame: Frame) -> (ExecuteResult, CodeReader) {
+
+    let Frame {
+        operand_stack,
+        local_vars,
+    } = frame;
+
+    let (v2, operand_stack) = operand_stack.pop_int();
+    let (v1, operand_stack) = operand_stack.pop_int();
+    let result = v1 & v2;
+    let operand_stack = operand_stack.push_int(result);
+
+    let frame = Frame {
+        operand_stack,
+        local_vars,
+    };
+    let execute_result = ExecuteResult { frame, offset: 0 };
+    (execute_result, code_reader)
+}
+
+#[allow(non_snake_case)]
+pub fn LAND(code_reader: CodeReader, frame: Frame) -> (ExecuteResult, CodeReader) {
+
+    let Frame {
+        operand_stack,
+        local_vars,
+    } = frame;
+
+    let (v2, operand_stack) = operand_stack.pop_long();
+    let (v1, operand_stack) = operand_stack.pop_long();
+    let result = v1 & v2;
+    let operand_stack = operand_stack.push_long(result);
+
+    let frame = Frame {
+        operand_stack,
+        local_vars,
+    };
+    let execute_result = ExecuteResult { frame, offset: 0 };
+    (execute_result, code_reader)
+}
+
+#[cfg(test)]
+mod tests {
+    use instruction::math::and::*;
+
+    use instruction::instruction::ExecuteResult;
+    use rtda::frame::Frame;
+    use rtda::local_vars::LocalVars;
+    use rtda::operand_stack::OperandStack;
+    use util::code_reader::CodeReader;
+
+    trait Push {
+        fn go_on(self, operand_stack : OperandStack) -> OperandStack;
+    }
+
+    impl Push for i64 {
+        fn go_on(self, operand_stack : OperandStack) -> OperandStack {
+            operand_stack.push_long(self)
+        }
+    }
+
+    impl Push for i32 {
+        fn go_on(self, operand_stack : OperandStack) -> OperandStack {
+            operand_stack.push_int(self)
+        }
+    }
+
+    #[test]
+    #[allow(non_snake_case)]
+    fn test_IAND() {
+        let frame = create_frame(3i32, 2i32);
+        let (ExecuteResult { frame, offset: _ }, _) = IAND(CodeReader::new(&vec![]), frame);
+        let (val, _) = frame.operand_stack.pop_int();
+        assert_eq!(val, 2);
+    }
+
+    #[test]
+    #[allow(non_snake_case)]
+    fn test_LAND() {
+        let frame = create_frame(3i64, 2i64);
+        let (ExecuteResult { frame, offset: _ }, _) = LAND(CodeReader::new(&vec![]), frame);
+        let (val, _) = frame.operand_stack.pop_long();
+        assert_eq!(val, 2i64);
+    }
+
+    fn create_frame<T : Push>(op1: T, op2: T) -> Frame {
+        let operand_stack = OperandStack::new(10);
+        let operand_stack = op1.go_on(operand_stack);
```

Comment:
They are equal just like `3.greater_than(5)` and `3 > 5` are.
The real reason to add it was to cut redundancy in our tests.

Reply:
I forgot to add to instruction :P

Reply:
I seriously thought about it and I tend to use 'push' directly rather than creating a 'Push Trait' because it may brings more reading burden than simplicity.
Do you agree with me?😄

Reply:
It is also possible that I did not understand your intentions.

Reply:
Sure. I suppose we can create a macro or something for our tests to become less cumbersome.
Let me remove the `Push` trait and re-write the test cases.

Reply:
Yes, I also think that there are too many boilerplate code in the test. I consider to try macro or something to improve this situation. And I reckon that If function works, No macro.